# Desafío: Estadística Descriptiva y Probabilidades
## Daniel Rojas López

En este notebook realizaremos un análisis estadístico descriptivo sobre salarios en Data Science y luego aplicaremos conceptos de probabilidad utilizando datos de la Copa del Mundo 2014.

In [2]:
import pandas as pd

# Cargar el dataset de salarios
df = pd.read_csv("worldcup2014.csv")
total = len(df)
total


32

In [3]:
df.head(5)
#df.head(total)

,team,continent,group,group_pos,cantidad_juegos,juegos_ganados,juegos_empatados,juegos_perdidos,goles_favor,goles_contra,goles_diferencia,puntos,clasificado
0,Brazil,southamerica,A,1,3,2,1,0,7,2,5,7,1
1,Mexico,northamerica,A,2,3,2,1,0,4,1,3,7,1
2,Croatia,europe,A,3,3,1,0,2,6,6,0,3,0
3,Cameroon,africa,A,4,3,0,0,3,1,9,−8,0,0
4,Netherlands,europe,B,1,3,3,0,0,10,3,7,9,1


In [4]:
condicionAfrica = df['continent'] == 'africa'
condicionAsia = df['continent'] == 'asia'
condicionEuropa = df['continent'] == 'europe'
condicionNorteamerica = df['continent'] == 'northamerica'
condicionSudamerica = df['continent'] == 'southamerica'
condicionClasificado = df['clasificado'] == 1
condicionGanados = df['juegos_ganados'] > 0


#print(condicionAfrica)

In [5]:
#pertenecer a África
p_Africa=condicionAfrica.sum() / total
print(f"pertenecer a África: {p_Africa:.3%}")

#pertenecer a Asia
p_Asia = condicionAsia.sum() / total
print(f"pertenecer a Asia: {p_Asia:.3%}")

#pertenecer a Europa
p_Europa = condicionEuropa.sum() / total
print(f"pertenecer a Europa: {p_Europa:.3%}")

#pertenecer a Norteamérica
p_Norteamerica= condicionNorteamerica.sum() / total
print(f"pertenecer a Norteamérica: {p_Norteamerica:.3%}")

#pertenecer a Sudamérica
p_Sudamerica = condicionSudamerica.sum() / total
print(f"pertenecer a Sudamérica: {p_Sudamerica:.3%}")

#clasificar a segunda ronda
p_Clasificado=condicionClasificado.sum() / total
print(f"Clasificar a segunda ronda: {p_Clasificado:.3%}")

#ganar al menos un partido
p_Ganados=condicionGanados.sum() / total
print(f"Ganar al menos un partido: {p_Ganados:.3%}")

#pertenecer a africa y clasificar a segunda ronda
p_AfricaClasificar=(condicionAfrica & condicionClasificado).sum() / total
print(f"pertenecer a africa y clasificar a segunda ronda: {p_AfricaClasificar:.3%}")

#pertenecer a Europa y clasificar a segunda ronda
p_EuropaClasificar=(condicionEuropa & condicionClasificado).sum() / total
print(f"pertenecer a Europa y clasificar a segunda ronda: {p_EuropaClasificar:.3%} ")

#pertenecer a Asia y clasificar a segunda ronda
p_AsiaClasificar=(condicionAsia & condicionClasificado).sum() / total
print(f"pertenecer a Asia y clasificar a segunda ronda: {p_AsiaClasificar:.3%} ")

#pertenecer a Sudamérica y clasificar a segunda ronda
p_SudamericaClasificar=(condicionSudamerica & condicionClasificado).sum() / total
print(f"pertenecer a Sudamérica y clasificar a segunda ronda: {p_SudamericaClasificar:.3%} ")

#pertenecer a Norteamérica y clasificar a segunda ronda
p_NorteamericaClasificar=(condicionNorteamerica & condicionClasificado).sum() / total
print(f"pertenecer a Norteamérica y clasificar a segunda ronda: {p_EuropaClasificar:.3%} ")


#pertenecer a África o clasificar a segunda ronda
p_AfricaoClasificar=(condicionAfrica | condicionClasificado).sum() / total
print(f"pertenecer a África o clasificar a segunda ronda: {p_AfricaoClasificar:.3%} ")

#pertenecer a Europa o clasificar a segunda ronda
p_EuropaoClasificar=(condicionEuropa | condicionClasificado).sum() / total
print(f"pertenecer a Europa o clasificar a segunda ronda: {p_EuropaoClasificar:.3%} ")

#pertenecer a Asia o clasificar a segunda ronda
p_AsiaoClasificar=(condicionAsia | condicionClasificado).sum() / total
print(f"pertenecer a Asia o clasificar a segunda ronda: {p_AsiaoClasificar:.3%} ")


#pertenecer a Sudamérica o clasificar a segunda ronda
p_SudamericaoClasificar=(condicionSudamerica | condicionClasificado).sum() / total
print(f"pertenecer a Sudamérica o clasificar a segunda ronda: {p_SudamericaoClasificar:.3%} ")

#pertenecer a Norteamérica o clasificar a segunda ronda
p_NorteamericaoClasificar=(condicionNorteamerica | condicionClasificado).sum() / total
print(f"pertenecer a Norteamérica o clasificar a segunda ronda: {p_NorteamericaoClasificar:.3%} ")

#pertenecer a África y haber ganado al menos un partido
p_AfricaGanar=(condicionAfrica & condicionGanados).sum() / total
print(f"pertenecer a África y ganado al menos un partido: {p_AfricaGanar:.3%} ")


#pertenecer a Europa y haber ganado al menos un partido
p_EuropaGanar=(condicionEuropa & condicionGanados).sum() / total
print(f"pertenecer a Europa y ganado al menos un partido: {p_EuropaGanar:.3%} ")


#pertenecer a Asia y haber ganado al menos un partido
p_AsiaGanar=(condicionAsia & condicionGanados).sum() / total
print(f"pertenecer a Asia y ganado al menos un partido: {p_AsiaGanar:.3%} ")


#pertenecer a Sudamérica y haber ganado al menos un partido
p_SudamericaGanar=(condicionSudamerica & condicionGanados).sum() / total
print(f"pertenecer a Sudamérica y ganado al menos un partido: {p_SudamericaGanar:.3%} ")

#pertenecer a Norteamérica y haber ganado al menos un partido
p_NorteamericaGanar=(condicionNorteamerica & condicionGanados).sum() / total
print(f"pertenecer a Norteamérica y ganado al menos un partido: {p_NorteamericaGanar:.3%} ")

#haber ganado al menos un partido y clasificar a segunda ronda
p_GanadoClasificado=(condicionGanados & condicionClasificado).sum() / total
print(f"haber ganado al menos un partido y clasificar a segunda ronda: {p_GanadoClasificado:.3%} ")


pertenecer a África: 15.625%
pertenecer a Asia: 12.500%
pertenecer a Europa: 40.625%
pertenecer a Norteamérica: 12.500%
pertenecer a Sudamérica: 18.750%
Clasificar a segunda ronda: 50.000%
Ganar al menos un partido: 71.875%
pertenecer a africa y clasificar a segunda ronda: 6.250%
pertenecer a Europa y clasificar a segunda ronda: 18.750% 
pertenecer a Asia y clasificar a segunda ronda: 0.000% 
pertenecer a Sudamérica y clasificar a segunda ronda: 15.625% 
pertenecer a Norteamérica y clasificar a segunda ronda: 18.750% 
pertenecer a África o clasificar a segunda ronda: 59.375% 
pertenecer a Europa o clasificar a segunda ronda: 71.875% 
pertenecer a Asia o clasificar a segunda ronda: 62.500% 
pertenecer a Sudamérica o clasificar a segunda ronda: 53.125% 
pertenecer a Norteamérica o clasificar a segunda ronda: 53.125% 
pertenecer a África y ganado al menos un partido: 9.375% 
pertenecer a Europa y ganado al menos un partido: 34.375% 
pertenecer a Asia y ganado al menos un partido: 0.000% 
pertenecer 

In [6]:
#pertenecer a Europa, si se sabe que clasificó
p_Europa_Dado_Clasificado = p_EuropaClasificar / p_Clasificado
print(f"pertenecer a Europa, si se sabe que clasificó: {p_Europa_Dado_Clasificado:.3%} ")

#pertenecer a Sudamérica, si se sabe que clasificó
p_Sudamerica_Dado_Clasificado = p_SudamericaClasificar / p_Clasificado
print(f"pertenecer a Sudamérica, si se sabe que clasificó: {p_Sudamerica_Dado_Clasificado:.3%} ")

#pertenecer a África, si se sabe que ganó al menos un partido

p_Africa_Dado_Ganado = p_AfricaGanar / p_Ganados
print(f"pertenecer a África, si se sabe que ganó al menos un partido: {p_Africa_Dado_Ganado:.3%} ")

#haber ganado al menos un partido, si se sabe que clasificó

p_Ganado_Dado_Clasificado = p_GanadoClasificado / p_Clasificado
print(f"haber ganado al menos un partido, si se sabe que clasificó: {p_Ganado_Dado_Clasificado:.3%} ")

#haber ganado al menos un partido, si se sabe que pertenece a Norteamérica

p_Ganado_Dado_Norteamerica = p_NorteamericaGanar / p_Norteamerica
print(f"haber ganado al menos un partido, si se sabe que pertenece a Norteamérica: {p_Ganado_Dado_Norteamerica:.3%} ")

#NO haber clasificado, si se sabe que pertenece a África

p_NoClasificado_Dado_Africa = 1 - (p_AfricaClasificar / p_Africa)
print(f"NO haber clasificado, si se sabe que pertenece a África: {p_NoClasificado_Dado_Africa:.3%} ")


pertenecer a Europa, si se sabe que clasificó: 37.500% 
pertenecer a Sudamérica, si se sabe que clasificó: 31.250% 
pertenecer a África, si se sabe que ganó al menos un partido: 13.043% 
haber ganado al menos un partido, si se sabe que clasificó: 100.000% 
haber ganado al menos un partido, si se sabe que pertenece a Norteamérica: 75.000% 
NO haber clasificado, si se sabe que pertenece a África: 60.000% 


#interpreta y aplica la probabilidad condicional y el teorema de Bayes



In [7]:
#Teorema de Bayes - Ejemplo práctico:
#Problema: Si un equipo clasificó, ¿cuál es la probabilidad de que sea europeo?

#Por Bayes: 
#P(Europa|Clasificado) = [P(Clasificado|Europa) × P(Europa)] / P(Clasificado)

# Método directo (condicional)
p_Europa_Dado_Clasificado_directo = p_EuropaClasificar / p_Clasificado
 
# Método por Bayes
p_Clasificado_Dado_Europa = p_EuropaClasificar / p_Europa
p_Europa_Dado_Clasificado_bayes = (p_Clasificado_Dado_Europa * p_Europa) / p_Clasificado



#Interpretación de resultados:
P(Europa|Clasificado) = 37.5%: De los equipos que clasificaron, 37.5% eran europeos
P(Clasificado|Europa) = 46.15%: De los equipos europeos, 46.15% clasificaron
P(Ganó|Clasificado) = 100%: Todos los equipos que clasificaron ganaron al menos un partido


#Verificación del Teorema de Bayes:
#Ambos métodos deben dar el mismo resultado, demostrando que:

P(Europa|Clasificado) = P(Europa ∩ Clasificado) / P(Clasificado) 
                      = [P(Clasificad